protocol/contracts/amo/test/goerli/RamosGoerli.sol

```diff
@@ -0,0 +1,410 @@
+    ) external onlyOwner {
+        // validate request
+        if (request.assets.length != request.maxAmountsIn.length || 
+            request.assets.length != 2 || 
+            request.fromInternalBalance == true) {
+                revert AMOCommon.InvalidBalancerVaultRequest();
+        }
+
+        uint256 templeAmount = request.maxAmountsIn[templeBalancerPoolIndex];
+        AMO__ITempleERC20Token(address(temple)).mint(address(this), templeAmount);
+        // safe allowance stable and TEMPLE
+        temple.safeIncreaseAllowance(address(balancerVault), templeAmount);
+
+        // join pool
+        uint256 bptAmountBefore = bptToken.balanceOf(address(this));
+        balancerVault.joinPool(balancerPoolId, address(this), address(this), request);
+        uint256 bptAmountAfter = bptToken.balanceOf(address(this));
+        uint256 bptIn;
+        unchecked {
+            bptIn = bptAmountAfter - bptAmountBefore;
+        }
+        if (bptIn < minBptOut) {
+            revert AMOCommon.InsufficientAmountOutPostcall(minBptOut, bptIn);
+        }
+
+        emit AddedLiquidity(request.maxAmountsIn[0], request.maxAmountsIn[1], bptIn);
+    }
+
+    /**
+     * @notice Remove liquidity from balancer pool receiving both TEMPLE and stable tokens from balancer pool. 
+     * Treasury Price Floor is expected to be within bounds of multisig set range.
+     * Withdraw and unwrap BPT tokens from Aura staking and send to balancer pool to receive both tokens.
+     * @param request Request for use in balancer pool exit
+     */
+    function removeLiquidity(
+        AMO__IBalancerVault.ExitPoolRequest memory request
+    ) external onlyOwner {
+        // validate request
+        if (request.assets.length != request.minAmountsOut.length || 
+            request.assets.length != 2 || 
+            request.toInternalBalance == true) {
+                revert AMOCommon.InvalidBalancerVaultRequest();
+        }
+
+        uint256 templeAmountBefore = temple.balanceOf(address(this));
+        uint256 stableAmountBefore = stable.balanceOf(address(this));
+
+        // use contract bpt. requires contract has enough bpt tokens
+
+        balancerVault.exitPool(balancerPoolId, address(this), address(this), request);
+
+        // validate amounts received
+        uint256 receivedAmount;
+        for (uint i=0; i<request.assets.length; ++i) {
+            if (request.assets[i] == address(temple)) {
+                unchecked {
+                    receivedAmount = temple.balanceOf(address(this)) - templeAmountBefore;
+                }
+                if (receivedAmount > 0) {
+                    AMO__ITempleERC20Token(address(temple)).burn(receivedAmount);
+                }
+            } else if (request.assets[i] == address(stable)) {
+                unchecked {
+                    receivedAmount = stable.balanceOf(address(this)) - stableAmountBefore;
+                }
+            }
+
+            // revert if insufficient amount received
```

Comment:
What's the tl;dr why you need a separate goerli contract? Going to be annoying to keep changes in sync -- eg you removed this check from the mainnet version

Reply:
Goerli version has staking contract  and aura interactions removed. It was impossible to add pool on aura goerli because the poolmanager function was protected.
For the code removed, this is unreachable code because balancer checks the minimum amounts